stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/utils/extensions/Channel.kt

```diff
@@ -0,0 +1,69 @@
+package io.getstream.chat.android.ui.utils.extensions
+
+import android.content.Context
+import android.graphics.drawable.Drawable
+import androidx.core.content.ContextCompat
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.ChannelUserRead
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.client.models.name
+import io.getstream.chat.android.livedata.ChatDomain
+import io.getstream.chat.android.ui.R
+import java.util.Date
+
+internal fun Channel.getUsers(excludeCurrentUser: Boolean = true): List<User> =
+    members
+        .map { it.user }
+        .let { users ->
+            when {
+                excludeCurrentUser -> users.withoutCurrentUser()
+                else -> users
+            }
+        }
+
+internal fun Channel.getDisplayName(): String = name.takeIf { it.isNotEmpty() }
+    ?: getUsers().joinToString { it.name }
+
+internal fun Channel.getLastMessage(): Message? =
+    messages
```

Comment:
three times filtering can be expensive. We can use Kotlin sequences for this, it gives you some optimisation on big amount of channel's messages.